### DISCUSSION TITLE
⭐ Xcode 16.0 will be default starting October 6

### DISCUSSION THREAD

**Discussion by helinanever** [codemagic-ci-cd/codemagic-docs#2749] [Announcements]
As of **October 6, 2024**, we are introducing the Xcode 16.0 build machine image as our new default. Additionally, Xcode `latest` will be switched to `16.0`.

Key points:
- If your Xcode version is set to `latest` or if no specific Xcode version is configured, then your builds will utilize the Xcode 16.0 image.
- Refer to the [build machine specs](https://docs.codemagic.io/specs/versions-macos/) for details on preinstalled software and available runtimes.

Action required:
- Please review your build configuration promptly to avoid unexpected behavior.
- You can always set a specific Xcode version in your build settings. Configure this in the Build section of the Flutter workflow editor or the environment section of codemagic.yaml.

If you have any questions or concerns, don't hesitate to get in touch with us.

**Comment by orestesgaolin** [comment]
We started to experience issues with AppStore Connect. This is due to 3rd party dependencies on iOS that are still using bitcode. One solution is to stay on xcode 15.4, the other is stripping bitcode from the 3rd party framework ([example article](https://medium.com/@abdulahad2024/fixing-bitcode-issues-in-xcode-16-how-to-resolve-invalid-executable-errors-when-uploading-ios-da07a5a39c7c)).

```
UserInfo={status=409, detail=Invalid Executable. The executable 'Runner.app\/Frameworks\/SprenCore.framework\/SprenCore' contains bitcode., id=b2889d78-513b-491e-947b-4fadef98e7bc, code=STATE_ERROR.VALIDATION_ERROR.90482, title=Asset validation failed, NSLocalizedFailureReason=Invalid Executable. The executable 'Runner.app\/Frameworks\/SprenCore.framework\/SprenCore' contains bitcode., NSLocalizedDescription=Asset validation failed}","NSLocalizedDescription":"Asset validation failed","iris-code":"STATE_ERROR.VALIDATION_ERROR.90482","NSLocalizedFailureReason":"Invalid Executable. The executable 'Runner.app\/Frameworks\/SprenCore.framework\/SprenCore' contains bitcode.
```

**Reply by mikhail-tokarev** [reply]
thank you for sharing this info.

**Comment by catarinaguerreiro** [comment]
Hello! When Xcode 16 stated to be used in codemagic I started to have problems in the codemagic build (but it works fine locally on my computer with Xcode 16). 

I'm having this error:

```
Failed to archive ios/HealthHive.xcworkspace. The following build commands failed:

CompileC /Users/builder/Library/Developer/Xcode/DerivedData/HealthHive-bgifpfychmrlnvdtvzhupyqnroyo/Build/Intermediates.noindex/ArchiveIntermediates/HealthHive/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/Rollbar.build/Objects-normal/arm64/KSMachineContext.o /Users/builder/clone/ios/Pods/Rollbar/KSCrash/Source/KSCrash/Recording/Tools/KSMachineContext.c normal arm64 c com.apple.compilers.llvm.clang.1_0.compiler (in target 'Rollbar' from project 'Pods')
	...
	note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'DatadogInternal' from project 'Pods')
	note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'DatadogCrashReporting-DatadogCrashReporting' from project 'Pods')
	note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'DatadogCrashReporting' from project 'Pods')
	note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'DatadogCore-DatadogCore' from project 'Pods')
	note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'AppAuth-AppAuthCore_Privacy' from project 'Pods')
	note: Disabling previews because SWIFT_VERSION is set and SWIFT_OPTIMIZATION_LEVEL=-O, expected -Onone (in target 'AppAuth' from project 'Pods')
```
	
Already searched but can't find any post or discussion related with that, can anyone help me on that? 

Thank you in advance

**Reply by dtrdic** [reply]
Hello @catarinaguerreiro i already responded on [Discord](https://discord.com/channels/1131597315707261018/1293888598135013499/1295369228072910939) but i'll share it here as well.

Please try to clean your project locally by removing the `podfile.lock` and `package-lock.json` files.
These files can sometimes lock dependencies to specific versions that may no longer be compatible with the latest xcode version or your current configuration. After removing them, please run locally:
```
yarn install
cd ios && pod install
```
to regenerate those files.

Also, in your codemagic.yaml, you have set `xcode: latest`, which always uses the latest Xcode version. This can lead to issues like the one you're experiencing. I recommend hardcoding the Xcode version you're using locally, for example:

`xcode: 15.4`
